openmls/src/credentials/mod.rs

```diff
@@ -287,48 +286,53 @@ impl Credential {
 ///
 /// OpenMLS provides an implementation of signature keys for convenience in the
 /// `openmls_basic_credential` crate.
-#[derive(
-    Debug,
-    Clone,
-    PartialEq,
-    Eq,
-    Serialize,
-    Deserialize,
-    TlsSerialize,
-    TlsDeserialize,
-    TlsDeserializeBytes,
-    TlsSize,
-)]
+#[derive(Debug, Clone, PartialEq, Eq, Serialize, Deserialize)]
 pub struct BasicCredential {
-    credential: Credential,
+    header: Vec<u8>,
```

Comment:
Is there a reason not use `VLBytes` for these two? That would be more efficient.

Reply:
Yes, the reason is that I want to do the serialization in the constructor, otherwise we have to suppress a potential error in `impl From<BasicCredential> for Credential` and the whole point of this PR was to get rid of unwraps.